### PR TITLE
Improve android sample

### PR DESCRIPTION
**Pull request by hannesa2**
It doesn't build out of the box

* enable build on command line
* Remove an unused and error causing class
* Update to forced necessary Gradle version
* Give project a meaningful name

image

### CODE REVIEW

**Comment by gbaned** [comment]
@caisq, Can you please take a look this PR ? Thanks!

**Comment by caisq** [comment]
@gbaned I don't think I have the necessary background and expertise to review this PR. Can you check with experts such as @petewarden or @miaout17 ?

**Comment by hannesa2** [comment]
The breaking change was to delete this file 
`tensorflow/java/src/main/java/org/tensorflow/op/core/Zeros.java` 
It was preventing form successful build

**Comment by hannesa2** [comment]
` ./gradlew clean build`

show now

image

**Review by frankchn — APPROVED** [review]

**Comment by frankchn** [comment]
@hannesa2 What is the build error when `Zeros.java` is left alone? We do have users who are using code in `Zeros.java` so we can't delete it.

File: tensorflow/java/src/main/java/org/tensorflow/op/core/Zeros.java [hunk 1]
```diff
@@ -1,68 +0,0 @@
-/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
```
Comment (frankchn) [inline tensorflow/java/src/main/java/org/tensorflow/op/core/Zeros.java:1]: We checked and we have internal users using this code, so we can't delete it.

**Review by frankchn — CHANGES_REQUESTED** [review]

**Comment by hannesa2** [comment]
> @hannesa2 What is the build error when `Zeros.java` is left alone? We do have users who are using code in `Zeros.java` so we can't delete it.

@frankchn 
## A: This is the first error

image

## B: After solving A

image

Because I don't know your internal structure, I can't solve it to fulfill internal and external build

**Comment by frankchn** [comment]
@hannesa2 Have you tried importing `org.tensorflow.op.core.Fill` to `Zeros.java` to resolve that error instead?

**Comment by frankchn** [comment]
Note that `Fill` is a generated class, so you might have to run some generator in your environment in order for that symbol to resolve. I suspect you have to change/add gradle settings for that to work.

**Comment by hannesa2** [comment]
> org.tensorflow.op.core.Fill

Still the same
image

**Comment by frankchn** [comment]
There is a test for `Zeros.java` at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/test/java/org/tensorflow/op/core/ZerosTest.java so you might want to make sure the test can pass too.

**Comment by hannesa2** [comment]
> There is a test for Zeros.java at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/test/java/org/tensorflow/op/core/ZerosTest.java so you might want to make sure the test can pass too.

Sorry, it's out of the Android project 

image

**Comment by frankchn** [comment]
I think you have to somehow get gradle/your environment to run the code at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/gen/cc/op_generator.cc during builds. That code generates all the relevant op files that are used. The standard TensorFlow Java Bazel build environment is set up to do this, but I am not sure how to do it on gradle and android.

> Sorry, it's out of the Android project

Yeah it is, but it shows that the `Zeros.java` code can be built and runs properly and this seems to be more of ab build environment setup issue on the Android example.

**Comment by hannesa2** [comment]
> I think you have to somehow get gradle/your environment to run the code at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/gen/cc/op_generator.cc during builds.

I've the feeling that it's a way in the wrong direction.
I should run a generator, to fix a class, which Android don't need. This sounds not logic

**Comment by hannesa2** [comment]
And the CI is successful for Android ?

I guess at least for Android the command `./gradlew build` would be the right command

image

**Comment by frankchn** [comment]
> I've the feeling that it's a way in the wrong direction. I should run a generator, to fix a class, which Android don't need. This sounds not logic

Deleting a file that builds and tests in the normal TensorFlow environment with the standard TensorFlow build tooling is correct either.

> I guess at least for Android the command ./gradlew build would be the right command. This one will fail

ZerosTest.java will fail if you delete Zeros.java in the main TensorFlow build, and this will break the main TensorFlow Java build.

**Comment by hannesa2** [comment]
Ok, I keep the `Zeros.java` untouched. 
As long, as I don't really understand why it's included, I'm unable to fix this.

The Android CI builds now properly (but on local machine it fails because of untouched `Zeros.java`)

Now the pull request only fix the Gradle setup and update to recent Android Studio build tools

**Comment by frankchn** [comment]
Thanks @hannesa2! Just one comment: for the `tensorflow/examples/android/gradlew` file, can we not have the chmod set to 0755 and leave it as 0644?

**Comment by hannesa2** [comment]
> Thanks @hannesa2! Just one comment: for the `tensorflow/examples/android/gradlew` file, can we not have the chmod set to 0755 and leave it as 0644?

Really ? If I don't do it, I can't run it locally 

image

That's why I did ` chmod +x gradlew`

Short: At least for me it's necessary ! And when  you ever build it on CI, you will need it too

**Review by frankchn — APPROVED** [review]

**Pull request merged**